code/game/objects/items/tools/maintenance_tools.dm

```diff
@@ -528,8 +531,13 @@
 	cell = null
 	playsound(user, 'sound/machines/click.ogg', 20, 1, 5)
 	balloon_alert(user, "Removes the cell")
-	icon_state = "handheldcharger_black_empty"
+	update_icon_state()
 
 /obj/item/tool/handheld_charger/Destroy()
 	QDEL_NULL(cell)
 	return ..()
+
+/obj/item/tool/handheld_charger/hicapcell/Initialize(mapload)
+	. = ..()
+	cell = new /obj/item/cell/high(src)
+	update_icon_state()
```

Comment:
don't call this directly. call update_icon. Also you should probably do that at the charger level instead of this one.